tests/e2e/action_rapid_test.go

```diff
@@ -0,0 +1,491 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/require"
+	"pgregory.net/rapid"
+)
+
+// This file contains tests for serialization/deserialization of actions.
+// The tests are written using the rapid testing library, which allows us to
+// generate arbitrary actions and test that they can be serialized and
+// deserialized without error.
+// The generators for the various actions are defined in this file, and
+// essentially tell rapid how to build these actions.
+
+func TestActionMarshalling(t *testing.T) {
+	rapid.Check(t, func(t *rapid.T) {
+		action := GetActionGen().Draw(t, "Action")
+		err := MarshalAndUnmarshalAction(action)
+		if err != nil {
+			t.Fatalf("error marshalling and unmarshalling action: %v", err)
+		}
+	})
+}
+
+func MarshalAndUnmarshalAction(action interface{}) error {
+	// wraps the action with a step, since it needs custom unmarshalling that is called by the step unmarshaller
+	step := Step{
+		Action: action,
+	}
+	jsonobj, err := json.Marshal(step)
+	if err != nil {
+		return fmt.Errorf("error marshalling action inside step: %v", err)
+	}
+
+	var got Step
+	err = json.Unmarshal(jsonobj, &got)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling action inside step: %v", err)
+	}
+
+	diff := cmp.Diff(step, got)
+	if diff != "" {
+		return fmt.Errorf("got (-), want (+): %v", diff)
+	}
+
+	return nil
+}
+
+// This needs to be adjusted manually when new actions are added and should
+// include generators for all actions that are mentioned in main.go/runStep.
+func GetActionGen() *rapid.Generator[any] {
+	return rapid.OneOf(
+		GetStartSovereignChainActionGen().AsAny(),
```

Comment:
What about `submitChangeRewardDenomsProposalAction`?

Reply:
submitChangeRewardDenomsProposalAction yes, good catch.
startConsumerEvidenceDetectorAction no, this version of the codebase does not have it.

Reply:
1168acab2